api/system/v1alpha1/zone_insight.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package kuma.system.v1alpha1;
+
+option go_package = "v1alpha1";
+
+import "google/protobuf/timestamp.proto";
+import "validate/validate.proto";
+
+// ZoneInsight defines the observed state of a Remote Kuma CP.
+message ZoneInsight {
+
+  // List of KDS subscriptions created by a given Remote Kuma CP.
+  repeated KDSSubscription subscriptions = 1;
+}
+
+// KDSSubscription describes a single KDS subscription
+// created by a Remote to the Global.
+// Ideally, there should be only one such subscription per Remote lifecycle.
+// Presence of multiple subscriptions might indicate one of the following
+// events:
+// - transient loss of network connection between Remote and Global Control
+// Planes
+// - Remote Kuma CP restarts (i.e. hot restart or crash)
+// - Global Kuma CP restarts (i.e. rolling update or crash)
+// - etc
+message KDSSubscription {
+
+  // Unique id per KDS subscription.
+  string id = 1 [ (validate.rules).string.min_len = 1 ];
+
+  // Global CP instance that handled given subscription.
+  string global_instance_id = 2 [ (validate.rules).string.min_len = 1 ];
+
+  // Time when a given Remote connected to the Global.
+  google.protobuf.Timestamp connect_time = 3
+      [ (validate.rules).timestamp.required = true ];
+
+  // Time when a given Remote disconnected from the Global.
+  google.protobuf.Timestamp disconnect_time = 4;
+
+  // Status of the KDS subscription.
+  KDSSubscriptionStatus status = 5 [ (validate.rules).message.required = true ];
+}
+
+// KDSSubscriptionStatus defines status of an KDS subscription.
+message KDSSubscriptionStatus {
+
+  // Time when status of a given KDS subscription was most recently updated.
+  google.protobuf.Timestamp last_update_time = 1;
+
+  // Total defines an aggregate over individual KDS stats.
+  KDSServiceStats total = 2;
+
+  KDSServiceStats mesh = 3;
+
+  KDSServiceStats ingress = 4;
+
+  KDSServiceStats circuitBreaker = 5;
+
+  KDSServiceStats faultInjection = 6;
+
+  KDSServiceStats healthCheck = 7;
+
+  KDSServiceStats trafficLog = 8;
+
+  KDSServiceStats trafficPermission = 9;
+
+  KDSServiceStats trafficRoute = 10;
+
+  KDSServiceStats trafficTrace = 11;
+
+  KDSServiceStats proxyTemplate = 12;
+
+  KDSServiceStats secret = 13;
```

Comment:
can we have a map here? typ to stats? It would be great to avoid yet another place to modify when you add a new entity.

Reply:
Yes, map is much better, thanks